### DISCUSSION TITLE
How to block URLs in Cefsharp browser

### DISCUSSION THREAD

**Discussion by Aravind-pro** [cefsharp/CefSharp#4645] [Q&A]
I have made a cef browser in winforms and i have found a way to block urls . 
![image](https://github.com/cefsharp/CefSharp/assets/78254185/cf7a6ec5-c617-49c2-a517-3f53602d69b7)

In this , if betonline.ag is opened , it should redirect to newUrl i set. It worked but still the site can be loaded in browser. 
How can i add more instances of the url i want to block in one block of code itself?. Can i add more urls in the if condition?

**Comment by amaitland** [comment]
Please no images of code see https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#no-images-of-code

Use `OnBeforeBrowse` to cancel navigations to urls.

```c#
public class CustomRequestHandler : CefSharp.Handler.RequestHandler
{
	protected override bool OnBeforeBrowse(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, IRequest request, bool userGesture, bool isRedirect)
	{
                // Perform Url validation here.
		if (request.Url == "SomeUrl")
		{
                        //Do something here if required e.g. LoadUrl
			//Return true tocancel the request
			return true;
		}
		return base.OnBeforeBrowse(chromiumWebBrowser, browser, frame, request, userGesture, isRedirect);
	}
}

chromiumWebBrowser.RequestHandler = new CustomRequestHandler();
```

> How can i add more instances of the url i want to block in one block of code itself?. Can i add more urls in the if condition?

If your new to `C#` and need some guidance then try stackoverflow or similar. Your just comparing strings.
